Make nozzle buffer size configurable via manifest

knyar opened the pull request:
Also, rename `batch_count` and `batch_duration` internally to make it more obvious that they only apply to Stackdriver logging.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cloudfoundry-community/stackdriver-tools/138)
<!-- Reviewable:end -->

johnsonj commented:
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[jobs/stackdriver-nozzle/spec, line 39 at r1](https://reviewable.io:443/reviews/cloudfoundry-community/stackdriver-tools/138#-Kxtuha5culysYFY--gQ:-Kxtuha5culysYFY--gR:bloxvqo) ([raw file](https://github.com/cloudfoundry-community/stackdriver-tools/blob/de73cc72f7d4fdd09facb0eec8f15e9dd05c0afc/jobs/stackdriver-nozzle/spec#L39)):*
> ```
>     description: Contents of application_default_credentials.json, see https://cloud.google.com/logging/docs/agent/authorization#configuring_client_id_authorization.
> 
>   nozzle.metrics_buffer_duration:
> ```

Can you add these fields in [tile.yml.erb](https://github.com/cloudfoundry-community/stackdriver-tools/blob/develop/tile.yml.erb#L41) as well? 
Use the same name, description, default values as here for the forms section

Pipe the values through [here](https://github.com/cloudfoundry-community/stackdriver-tools/blob/develop/tile.yml.erb#L26) to the bosh manifest

---

*[src/stackdriver-nozzle/config/config.go, line 61 at r1](https://reviewable.io:443/reviews/cloudfoundry-community/stackdriver-tools/138#-KxtuBn5hmKotFWGZIdU:-KxtuBn5hmKotFWGZIdV:bxq3tvc) ([raw file](https://github.com/cloudfoundry-community/stackdriver-tools/blob/de73cc72f7d4fdd09facb0eec8f15e9dd05c0afc/src/stackdriver-nozzle/config/config.go#L61)):*
> ```Go
> 	// Stackdriver config
> 	ProjectID            string `envconfig:"gcp_project_id"`
> 	LoggingBatchCount    int    `envconfig:"batch_count" default:"10"`
> ```

update envconfig annotations here and for `batch_duration`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cloudfoundry-community/stackdriver-tools/138)*
<!-- Sent from Reviewable.io -->

johnsonj commented:
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cloudfoundry-community/stackdriver-tools/138)*
<!-- Sent from Reviewable.io -->

knyar commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/stackdriver-nozzle/config/config.go, line 61 at r1](https://reviewable.io:443/reviews/cloudfoundry-community/stackdriver-tools/138#-KxtuBn5hmKotFWGZIdU:-Kxu-H2twrPQOH7AKOgM:bx8o49w) ([raw file](https://github.com/cloudfoundry-community/stackdriver-tools/blob/de73cc72f7d4fdd09facb0eec8f15e9dd05c0afc/src/stackdriver-nozzle/config/config.go#L61)):*
<details><summary><i>Previously, johnsonj (Jeff Johnson) wrote…</i></summary><blockquote>

update envconfig annotations here and for `batch_duration`
</blockquote></details>

Any concerns about breaking existing users who might be setting variables with current names?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cloudfoundry-community/stackdriver-tools/138)*
<!-- Sent from Reviewable.io -->

knyar commented:
Review status: 2 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[jobs/stackdriver-nozzle/spec, line 39 at r1](https://reviewable.io:443/reviews/cloudfoundry-community/stackdriver-tools/138#-Kxtuha5culysYFY--gQ:-KxwV0alBr9NX9x4xgBg:b-896fix) ([raw file](https://github.com/cloudfoundry-community/stackdriver-tools/blob/de73cc72f7d4fdd09facb0eec8f15e9dd05c0afc/jobs/stackdriver-nozzle/spec#L39)):*
<details><summary><i>Previously, johnsonj (Jeff Johnson) wrote…</i></summary><blockquote>

Can you add these fields in [tile.yml.erb](https://github.com/cloudfoundry-community/stackdriver-tools/blob/develop/tile.yml.erb#L41) as well? 
Use the same name, description, default values as here for the forms section

Pipe the values through [here](https://github.com/cloudfoundry-community/stackdriver-tools/blob/develop/tile.yml.erb#L26) to the bosh manifest
</blockquote></details>

Done.

---

*[src/stackdriver-nozzle/config/config.go, line 61 at r1](https://reviewable.io:443/reviews/cloudfoundry-community/stackdriver-tools/138#-KxtuBn5hmKotFWGZIdU:-KxwWGJbZxB8U5AQlM65:bu0lrnh) ([raw file](https://github.com/cloudfoundry-community/stackdriver-tools/blob/de73cc72f7d4fdd09facb0eec8f15e9dd05c0afc/src/stackdriver-nozzle/config/config.go#L61)):*
<details><summary><i>Previously, knyar (Anton Tolchanov) wrote…</i></summary><blockquote>

Any concerns about breaking existing users who might be setting variables with current names?
</blockquote></details>

I guess no concerns since these are not configurable via tile properties or manifest. Renamed, updated README.md.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cloudfoundry-community/stackdriver-tools/138)*
<!-- Sent from Reviewable.io -->

johnsonj commented:
LGTM

---

Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cloudfoundry-community/stackdriver-tools/138#-:-KxxWaoowMrtjQNi-rA4:bb6ywmx)*
<!-- Sent from Reviewable.io -->
